### PR TITLE
Datamodel attr caching

### PR DESCRIPTION
**Pull request by mkundu1**
1. Cache datamodel attribute values in client - some changes are pending to unsubscribe the on-attribute-changed listeners on deleting named objects.
2. `add_on_deleted` method at datamodel objects.

### CODE REVIEW

File: src/ansys/fluent/core/services/datamodel_se.py [hunk 1]
```diff
@@ -337,6 +376,25 @@ def set_state(self, state: Any = None, **kwargs) -> None:
...
+        request.attribute = attrib
+        response = self.service.get_attribute_value(request)
+        return _convert_variant_to_value(response.result)
+
+    def _get_cached_attr(self, attrib: str) -> Any:
+        cached_val = self.cached_attrs.get(attrib)
+        if cached_val is None:
+            cached_val = self._get_remote_attr(attrib)
+            self.cached_attrs[attrib] = cached_val
+            self.add_on_attribute_changed(
```
Comment (mkundu1) [inline src/ansys/fluent/core/services/datamodel_se.py:392]: TODO: unsubscribe on named-object deletion

File: src/ansys/fluent/core/services/datamodel_se.py [hunk 2]
```diff
@@ -1138,6 +1134,20 @@ def __del__(self):
             # "Cannot invoke RPC on closed channel!"
             pass
 
+    # Currently getting an error whenever I'm calling add_on_command_attribute_changed on self or parent
+    # def _get_cached_attr(self, attrib: str) -> Any:
+    #     cached_val = self.cached_attrs.get(attrib)
+    #     if cached_val is None:
+    #         cached_val = self._get_remote_attr(attrib)
+    #         parent = PyStateContainer(self.service, self.rules, self.path[0:-1])
+    #         parent.add_on_command_attribute_changed(
```
Comment (mkundu1) [inline src/ansys/fluent/core/services/datamodel_se.py:1143]: Getting an error from server with either
```
eventrequest {
  rules: "meshing"
  commandAttributeChangedEventRequest {
    path: "/ImportGeometry:ImportGeometry2"
    command: "ImportGeometry"
    attribute: "FileName/isActive"
  }
}
```
or
```
eventrequest {
  rules: "meshing"
  commandAttributeChangedEventRequest {
    command: "ImportGeometry"
    attribute: "FileName/isActive"
  }
}
```
requests.
We have unittest with command-level attributes.

Comment (mkundu1) [reply src/ansys/fluent/core/services/datamodel_se.py]: command argument attributes won't be cached as they depends on command state in client, not entirely depends on server state

**Comment by seanpearsonuk** [comment]
@prmukherj we need one more reviewer for this PR. `add_on_deleted` will be useful for me.

**Review by prmukherj — APPROVED** [review]

**Pull request merged**